src/coreclr/jit/codegenxarch.cpp

```diff
@@ -10882,7 +10896,7 @@ void CodeGen::genVzeroupperIfNeeded(bool check256bitOnly /* = true*/)
 
     if (emitVzeroUpper)
     {
-        assert(compiler->canUseVexEncoding());
+        assert(compiler->canUseVexEncoding() || compiler->canUseEvexEncoding());
```

Comment:
Shouldn't the former assert alone be enough?

We will never have a case where `canUseEvex` is true but `canUseVex` is false.